src/manifest.js

```diff
@@ -0,0 +1,140 @@
+import fs from 'fs';
+import * as utils from './utils/index.js';
+import zlib from 'zlib';
+import { promisify } from 'util';
+import AWS from 'aws-sdk';
+
+const gunzip = promisify(zlib.gunzip)
+
+const LATEST = "LATEST"
+
+const {ResourceVersions} = (() => {
+
+  const data = {};
+
+  update(data); // async, but we don't want to block
+
+  // FIXME - use setInterval _or_ expose endpoint to run update()
+
+  class ResourceVersions {
+    constructor(resourceType, resourceId) {
+      this.data = {};
+      if (!data[resourceType]) {
+        utils.warn(`Attempted to access resources of type ${resourceType} but there are none in the manifest`);
+        return;
+      }
+      this.data = data[resourceType][resourceId] || {};
+    }
+
+    /**
+     * Given a YYYY-MM-DD string return one of four values:
+     * - LATEST: if the date is more recent that any in the manifest
+     * - false: there are no records in the manifest, or the dateDescriptor
+     *          is mis-formed
+     * - YYYY-MM-DD string: the closest match (may be an exact match)
+     * 
+     * In the future this will be expanded to parse other kinds of
+     * dateDescriptors (e.g. a descriptor meaning "6 months ago").
+     * 
+     * When there is not an exact match we interpret closest to be
```

Comment:
@corneliusroemer had the good suggestion to invert this so that for a given date we return the version that was the current at that time.

Reply:
That makes much more sense to me than "closest". It also matches how I've thought about this in terms of answering the question, "What would I have seen for this had I looked on date X?"